### PR TITLE
Helm test

### PR DESCRIPTION
**Pull request by paulmurdoch19**
configured gen3-Helm to work with pcdc images of services that currently are supported in gen3-helm

### CODE REVIEW

File: helm/gen3/Chart.yaml [hunk 1]
```diff
@@ -87,12 +87,19 @@ dependencies:
   version: "0.1.10"
   repository: "file://../wts"
   condition: wts.enabled
-
+- name: pcdcanalysistools
+  version: "0.1.0"
+  repository: "file://../pcdcanalysistools"
```
Comment (grugna) [inline helm/gen3/Chart.yaml:92]: What is this version?

File: helm/gen3/values.yaml [hunk 2]
```diff
@@ -106,6 +106,9 @@ aws-es-proxy:
     # -- (str) AWS secret access key for aws-es-proxy
     awsSecretAccessKey: ""
 
+elasticsearch:
+  enabled: false
```
Comment (grugna) [inline helm/gen3/values.yaml:110]: Why is ES disabled?

File: helm/pcdcanalysistools/pcdcanalysistools-secret/settings.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
+from PcdcAnalysisTools.api import app, app_init
```
Comment (grugna) [inline helm/pcdcanalysistools/pcdcanalysistools-secret/settings.py:1]: [This](https://github.com/chicagopcdc/cloud-automation/blob/pcdc_dev/apis_configs/pcdcanalysistools_settings.py) is the up to date version

File: helm/portal/templates/deployment.yaml [hunk 4]
```diff
@@ -74,7 +74,7 @@ spec:
             {{- include "common.datadogEnvVar" . | nindent 12 }}
             {{- end }}
             - name: HOSTNAME
-              value: revproxy-service
+              value: portal-dev.pedscommons.org
```
Comment (grugna) [inline helm/portal/templates/deployment.yaml:77]: Figure out if you can revert this to a generic one instead of having to set up a fake URL

File: helm/portal/templates/deployment.yaml [hunk 5]
```diff
@@ -172,7 +172,7 @@ spec:
             # - name: BASENAME
           volumeMounts:
             - name: "config-volume"
-              mountPath: "/data-portal/data/config/gitops.json"
+              mountPath: "/data-portal/data/config/pcdc.json"
```
Comment (grugna) [inline helm/portal/templates/deployment.yaml:175]: Is this change necessary? Or is it only to connect to the local running portal?

File: values.yaml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+aws-es-proxy:
+  enabled: true
+  image:
+    repository: abutaha/aws-es-proxy
+    tag: 0.8
+
+#modify gen3 chart so that elasticsearch automaticcally starts
+elasticsearch:
+  enabled: false
```
Comment (grugna) [inline values.yaml:39]: We probably need t his otherwise guppy will not start. We can wait until we load some fake data tho, before enabling it.

File: values.yaml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+#     repository: fluent/fluentd-kubernetes-daemonset
+#     tag: v1.15.3-debian-cloudwatch-1.0
+
+
+
+
+# revproxy:
+#   image:
+#     repository: quay.io/cdis/nginx
+#     tag: 1.17.6-ctds-1.0.1
```
Comment (grugna) [inline values.yaml:156]: why rev proxy is commented out?

File: values.yaml [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+wts:
+  image:
+    repository: quay.io/cdis/workspace-token-service
+    tag: 2023.08
+
+# amanuensis:
+#     image:
+#       repository: quay.io/pcdc/amanuensis
+#       tag: "test"
```
Comment (grugna) [inline values.yaml:102]: why is amanuensis commented out?

File: helm/portal/defaults/gitops.json [hunk 9]
```diff
@@ -1,351 +1,344 @@
 {
-  "subcommons": [
-    {
-      "URL": "https://tb.diseasedatahub.org/",
-      "name": "TB"
-    },
-    {
-      "URL": "https://aids.diseasedatahub.org/",
```
Comment (grugna) [inline helm/portal/defaults/gitops.json:8]: The most up to date gitops.json is [here](https://github.com/chicagopcdc/configuration-files/blob/master/cloud/portal-dev.pedscommons.org/portal/gitops.json).

Didn't you say that changing the default file was the reason of portal not picking up the new changes? Was that this defaults/gitops.json or default.json?

File: values.yaml (hunk 8 above)
Comment (paulmurdoch19) [reply values.yaml:102]: that was part of a test I was trying out I will change that

File: helm/gen3/values.yaml (hunk 2 above)
Comment (paulmurdoch19) [reply helm/gen3/values.yaml:110]: when ES and aws-es-proxy are enabled this comes up on start up "* services "elasticsearch" already exists" but everything still works when both are on

File: helm/gen3/Chart.yaml (hunk 1 above)
Comment (paulmurdoch19) [reply helm/gen3/Chart.yaml:92]: its a refrence to the chart version for pcdcanalysistools

File: helm/amanuensis/amanuensis-google-creds/amanuensis_google_app_creds_secret.json
Comment (grugna) [inline helm/amanuensis/amanuensis-google-creds/amanuensis_google_app_creds_secret.json:1]: amanuensis doesn't need this file

File: helm/amanuensis/amanuensis-google-creds/amanuensis_google_storage_creds_secret.json
Comment (grugna) [inline helm/amanuensis/amanuensis-google-creds/amanuensis_google_storage_creds_secret.json:1]: amanuensis doesn't need this file

File: helm/amanuensis/logo/logo.svg
Comment (grugna) [inline helm/amanuensis/logo/logo.svg:1]: replace this with our logo, you can find it on box or inside the configuration file repo /{ENV}/portal/logo.svg or gitops.svg

File: helm/amanuensis/templates/hpa.yaml
Comment (grugna) [inline helm/amanuensis/templates/hpa.yaml:1]: what is this for? I know what the horizontal autoscaler is, I am mostly wondering why did you add it since fence for instance doesn't have it

File: helm/amanuensis/templates/deployment.yaml
Comment (grugna) [inline helm/amanuensis/templates/deployment.yaml:1]: not sure if that will break things, but to maintain consistency with the others can we rename this amanuensis-deployment.yaml

File: helm/amanuensis/templates/amanuensis-secret.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+data:
+{{ (.Files.Glob "amanuensis-google-creds/*").AsSecrets | indent 2 }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: amanuensis-google-storage-creds-secret
+type: Opaque
+data:
+{{ (.Files.Glob "amanuensis-google-creds/*").AsSecrets | indent 2 }}
```
Comment (grugna) [inline helm/amanuensis/templates/amanuensis-secret.yaml:23]: there are no Google crews in amanuensis

File: helm/amanuensis/templates/hpa.yaml
Comment (paulmurdoch19) [reply helm/amanuensis/templates/hpa.yaml:1]: that is autogenerated from helm, im not sure why the cdis people removed it from fence but it seems to be present in the other services. Should I remove it?

File: helm/amanuensis/templates/deployment.yaml
Comment (paulmurdoch19) [reply helm/amanuensis/templates/deployment.yaml:1]: I think that fence is the only one that isnt named deployment.yaml because there is the presigned-urls deployment also in that chart

File: helm/amanuensis/templates/amanuensis-creds.yaml
Comment (grugna) [inline helm/amanuensis/templates/amanuensis-creds.yaml:1]: As mentioned the other day [these are the config](https://github.com/chicagopcdc/cloud-automation/blob/ab546ca2451fc273200893622f35c614d931353f/tf_files/shared/modules/k8s_configs/creds.tpl#L13C1-L22) expected by amanuensis.

File: helm/gen3/values.yaml (hunk 2 above)
Comment (grugna) [reply helm/gen3/values.yaml:110]: Sorry, I am not sure I understand.

File: helm/amanuensis/templates/hpa.yaml
Comment (grugna) [reply helm/amanuensis/templates/hpa.yaml:1]: If it is autogenerated you can leave it.

File: helm/portal/templates/deployment.yaml (hunk 5 above)
Comment (paulmurdoch19) [reply helm/portal/templates/deployment.yaml:175]: I changed it back I thought it mattered but it actually doesnt

File: values.yaml (hunk 7 above)
Comment (paulmurdoch19) [reply values.yaml:156]: this image was the one from that list of images to use from cloud automation but throws this error "2023/09/28 21:13:47 [emerg] 1#1: unknown directive "js_import" in /etc/nginx/nginx.conf:68
nginx: [emerg] unknown directive "js_import" in /etc/nginx/nginx.conf:68"

Comment (paulmurdoch19) [reply values.yaml:156]: https://github.com/chicagopcdc/configuration-files/blob/eb9816feed6ae757595731fa337d2cddfbea8513/cloud/portal-dev.pedscommons.org/manifest.json#L200

File: helm/gen3/values.yaml (hunk 2 above)
Comment (paulmurdoch19) [reply helm/gen3/values.yaml:110]: actually nvm it doesn't matter they both can be turned on the error above was something to do with helm giving elasticsearch and aws-es-proxy the same service name

File: values.yaml (hunk 6 above)
Comment (paulmurdoch19) [reply values.yaml:39]: sounds good!

File: helm/amanuensis/templates/amanuensis-creds.yaml
Comment (paulmurdoch19) [reply helm/amanuensis/templates/amanuensis-creds.yaml:1]: are these values from amanuensis config file  AWS_CREDENTIALS:
  'DATA_DELIVERY_S3_BUCKET':
    aws_access_key_id: 'DATA_DELIVERY_S3_BUCKET_ACCESS_KEY'
    aws_secret_access_key: 'DATA_DELIVERY_S3_BUCKET_PRIVATE_KEY' correlate to "data_delivery_bucket": "${data-release-bucket_name}",
        "data_delivery_bucket_aws_key_id": "${amanuensis-bot_user_id}",
        "data_delivery_bucket_aws_access_key": "${amanuensis-bot_user_secret}"

Comment (paulmurdoch19) [reply helm/amanuensis/templates/amanuensis-creds.yaml:1]: and this is the bucket correct, DATA_DOWNLOAD_BUCKET: 'bucket1'

File: helm/portal/defaults/gitops.json (hunk 9 above)
Comment (paulmurdoch19) [reply helm/portal/defaults/gitops.json:8]: I think at the time I thought that but I think now it was because of a formating error values.yaml and helm not picking it up and falling back to the default values. I ended up just changing the default values to our version since it will be used for the pcdc portal but anyone can still change the gitops in the values.yaml

Comment (paulmurdoch19) [reply helm/portal/defaults/gitops.json:8]: also should I change the gitops-favicon an gitops-logo to the ones in the folder should shared above?

File: helm/portal/templates/deployment.yaml (hunk 4 above)
Comment (paulmurdoch19) [reply helm/portal/templates/deployment.yaml:77]: I fixed it, dictionary_url needed to be set as an env variable in helm

**Comment by grugna** [comment]
Make sure to push the changes after you make them so they can show up here.

File: values.yaml (hunk 7 above)
Comment (grugna) [reply values.yaml]: Try to use [this](https://github.com/chicagopcdc/compose-services/blob/87c28721c61f7d9ddb8c59b55b7e68443a5c619f/docker-compose.yml#L318) revproxy instead. just use the 2023.09 tag instead of that old 2022 one.

Comment (paulmurdoch19) [reply values.yaml]: that one worked!
